src/Symfony/Component/Lock/Factory.php

```diff
@@ -28,7 +28,7 @@ class Factory implements LoggerAwareInterface
 
     private $store;
 
-    public function __construct(StoreInterface $store)
+    public function __construct(PersistStoreInterface $store)
```

Comment:
This deprecated Factory class should be deleted here instead of being updated

Reply:
Will rebased when https://github.com/symfony/symfony/pull/32495 gets merged

Reply:
why making 2 separate PRs to remove deprecated features of the Lock component (which is not a huge component) ?

Reply:
I wanted to limit the code changes to only one reponsability, If you think that's too much I can merge both in one.

Reply:
"Remove deprecated Lock features" can be considered as one responsibility too (a less granular one).